src/spikeinterface/core/generate.py

```diff
@@ -465,9 +466,9 @@ def inject_some_duplicate_units(sorting, num=4, max_shift=5, ratio=None, seed=No
     return sorting_with_dup
 
 
-def inject_some_split_units(sorting, split_ids=[], num_split=2, output_ids=False, seed=None):
+def inject_some_split_units(sorting, split_ids: list, num_split=2, output_ids=False, seed=None):
     """ """
-    assert len(split_ids) > 0, "you need to provide some ids to split"
+
```

Comment:
@alejoe91 
The error was caused by the fact that np.array(0) behaves as `False`. I should know better to not play with the truthiness of containers. 

However, as in this case, you actually ask the split_ids to be non-empty I Just made the argument a required one.